### PR TITLE
Deenurp issue27

### PR DESCRIPTION
**Pull request by crosenth**
* ``deenurp rdp_extract_sequences`` now updates old tax_ids instead of dropping them (issue: 27)
* ``deenurp rdp_extract_sequences`` creates new column is_classified instead of dropping tax_ids of unclassified sequences (issue: 28)
* function tax_of_genbank no longer returns None if string 'uncultured bacterium' is in the organism name (issue: 29)

### CODE REVIEW

**Comment by crosenth** [comment]
Also includes updates for issue 28 and 29.

File: deenurp/subcommands/rdp_extract_genbank.py [hunk 1]
```diff
@@ -114,9 +133,12 @@ def action(a):
             rdp_lineage = ';'.join(record.annotations.get('taxonomy', []))
             rdp_lineage = rdp_lineage.replace('"', '')
 
+            is_class = is_classified(tax_id)
+            is_class &= 'uncultured bacterium' not in get_organism(record)
```
Comment (nhoffman) [inline deenurp/subcommands/rdp_extract_genbank.py]: Does checking for 'uncultured bacterium' in the organism name this have any value here if we have already filtered on the basis of tax_id?

File: deenurp/subcommands/rdp_extract_genbank.py [hunk 2]
```diff
@@ -106,7 +125,7 @@ def action(a):
         if a.header:
             header = ('version', 'seqname', 'tax_id', 'accession',
                       'description', 'length', 'ambig_count',
-                      'is_type', 'rdp_lineage')
+                      'is_type', 'rdp_lineage', 'is_classified')
```
Comment (nhoffman) [inline deenurp/subcommands/rdp_extract_genbank.py]: Let's change 'is_classified' to 'taxid_is_classified' to clarify that we are just looking at the tax_id here and not other elements of the the sequence record. This should also be clarified in the module's docstring.

File: deenurp/subcommands/rdp_sequence_filter.py [hunk 3]
```diff
@@ -19,33 +20,39 @@ def count_ambiguous(seq):
...
+        default='named.seqs.fasta', help='[default %(default)s]')
+    p.add_argument(
+        '--named-info',
+        default='named.seq_info.csv', help='[default %(default)s]')
+    p.add_argument(
+        '--unnamed-seqs',
+        default='unnamed.seqs.fasta', help='[default %(default)s]')
+    p.add_argument(
+        '--unnamed-info',
+        default='unnamed.seq_info.csv', help='[default %(default)s]')
```
Comment (nhoffman) [inline deenurp/subcommands/rdp_sequence_filter.py]: Add an option to save a list of discarded seq IDs

File: deenurp/subcommands/rdp_sequence_filter.py [hunk 4]
```diff
@@ -68,7 +74,7 @@ def action(a):
                     continue
                 accepted += 1
 
-                if info['tax_id'] != '':
+                if info['is_classified'] == 'True':
```
Comment (nhoffman) [inline deenurp/subcommands/rdp_sequence_filter.py]: remember that the name of the column 'is_classified' has changed in the input

File: deenurp/subcommands/rdp_extract_genbank.py [hunk 5]
```diff
@@ -70,8 +83,17 @@ def is_type(record):
     return '(T)' in record.description
```
Comment (nhoffman) [inline deenurp/subcommands/rdp_extract_genbank.py]: might as well just update this function to include search terms for type strains

**Pull request merged**